### ISSUE TITLE
dispmanx_vncserver: main.c:605: main: Assertion `image' failed.

### ISSUE THREAD

**Issue by estivalet** [hanzelpeter/dispmanx_vnc#28]
got this in retropie 4.5 Any tips?

**Comment by hanzelpeter** [comment]
This assert means that it cannot allocate enough memory for display buffer.
What is the resolution of display you want to show? And what RPI model is it?
Some other program taking a lot of memory?

**Comment by estivalet** [comment]
no, I found the issue right now, the hdmi cable was loose on the raspberry PI and so I think it was not generating a signal for HDMI image

**Comment by jakubiakdev** [comment]
I have the same issue, but I don't have any monitors connected. Someone from kodi forums said that this application might help if there is no monitor but I don't know if it is true

**Comment by hanzelpeter** [comment]
Hello.

Could you please add first lines of output when program is run. 

When it is something like 
Display size is 0 x 46446656

I saw this on some firmware version and I needed to change display ID from 0 to 8 in open dispmanx display. I also need to get rpi4 and check on it. There I need probably t o add display ID parameter.

**Comment by John-Duh** [comment]
Hello, 
I'm having the same issue I believe. 
Here's the error message: 
Open display[0]...
Display is 4 x 1053349172
Rotate is 67108864
dispmanx_vncserver: main.c:669: main: Assertion `image' failed.
Aborted

I tried adding -d option as mentioned in the help but this didn't change the output. 

This is on a raspberry pi 4 running retropie

**Comment by jose1711** [comment]
Same thing on RPi3 running latest OSMC.
```
./dispmanx_vncserver  -d 0
Open display[0]...
Display is 4 x 1053650496
Rotate is 67108864
dispmanx_vncserver: main.c:669: main: Assertion `image' failed.
Aborted
```
Tested with `-d` set to 0, 2, 3 and 7 - always the same. Maybe it's related to this?
```
$ /opt/vc/bin/tvservice -o  
/opt/vc/bin/tvservice is not supported when using the vc4-kms-v3d driver.
Similar features are available with standard linux tools
such as modetest from libdrm-tests.
```

**Comment by hanzelpeter** [comment]
Hello.

I think this is related to vc4-kms-v3d.
I also use raspi-config and use vc4-fkms-v3d driver.

**Comment by jose1711** [comment]
Thank you, you were right. Switching to `fkms` solved it. Maybe you could it add it to `README.md`?

**Comment by Andrew-J-Larson** [comment]
Got things working in RetroPie on the Pi 4, running headless.

Optional, but recommended (change resolution first):
- Run: `sudo raspi-config`
- Go to **Display Options** > **Resolution** > Select your preferred screen size

Required
- Edit the `/boot/config.txt` file, and uncomment `hdmi_drive=2` (this will allow the pi to properly stay in HDMI mode while running headless, no HDMI cable needs to be plugged in).

After doing the above, `dispmanx_vncserver` runs properly, and outputs display to VNC.
